### ISSUE TITLE
Not sure how to resolve MA0016

### ISSUE THREAD

**Issue by grosch-intl** [meziantou/Meziantou.Analyzer#752]
This code is triggering MA0016: 

```cs
public Task<Dictionary<Guid, KnownCollateral>> GetTrackedKnownCollateralAsync(CancellationToken cancellationToken) {
    return context.KnownCollateral
            .ToDictionaryAsync(x => x.Uid, cancellationToken);
}
```

I'm not sure how it wants me to resolve that though.  If I make it `IDictionary<Guid, KnownCollateral>>` it's a compiler error.

**Comment by meziantou** [comment]
If you need to apply the fix, I think you need to use await to let the compiler apply the implicit conversion

````c#
public async Task<IDictionary<Guid, KnownCollateral>> GetTrackedKnownCollateralAsync(CancellationToken cancellationToken) {
    return await context.KnownCollateral
            .ToDictionaryAsync(x => x.Uid, cancellationToken);
}
````
